### PR TITLE
fix docs for new clippy lint

### PR DESCRIPTION
**Pull request by apoelstra**
There are a bunch of doccomments whose first lines are (much) too long. Most of these are also difficult to understand and/or out-of-date. Just rewrite them all.

### CODE REVIEW

**Comment by Kixunil** [comment]
Concept ACK but it *also* fails CI. :)

File: secp256k1-sys/src/lib.rs [hunk 1]
```diff
@@ -119,10 +124,19 @@ impl SchnorrSigExtraParams {
...
+/// randomize computations to protect against sidechannel attacks. In the
+/// past it has contained precomputation tables to speed up crypto operations.
+///
+/// It should be assumed to be expensive to create and therefore should be
+/// reused when possible.
+///
+/// If you create one of these with `secp256k1_context_create` you must
+/// destroy it with `secp256k1_context_destroy`. (Failure to destroy it is
+/// a memory leak; destroying it using any other allocator is likely to be
+/// undefined behavior.)
```
Comment (Kixunil) [inline secp256k1-sys/src/lib.rs:139]: I'd just write "is undefined behavior". I don't think it's useful to try to define some crazy circumstances when it's not.

File: src/ellswift.rs [hunk 2]
```diff
@@ -280,10 +281,15 @@ impl ElligatorSwiftSharedSecret {
     pub const fn as_secret_bytes(&self) -> &[u8; 32] { &self.0 }
 }
 
-/// Represents which party we are in the ECDH, A is the initiator, B is the responder.
-/// This is important because the hash of the shared secret is different depending on which party
-/// we are. In this context, "we" means the party that is using this library, and possesses the
-/// secret key passed to `ElligatorSwift::shared_secret`.
+/// Represents which party we are in the ECDH.
+///
+/// Here `A` is the initiator and `B` is the responder.
```
Comment (Kixunil) [inline src/ellswift.rs:286]: I guess we should've named the enum variants as such. :(

**Comment by apoelstra** [comment]
strace shows that xargo is failing trying to access some internal lockfile:

```
/store/home/apoelstra/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/Cargo.lock
```

Looks to me like xargo is just broken and I need to revert the update to the nightly compiler.

File: src/lib.rs [hunk 3]
```diff
@@ -194,9 +194,13 @@ use crate::ffi::CPtr;
...
-/// Trait describing something that promises to be a 32-byte random number; in particular,
-/// it has negligible probability of being zero or overflowing the group order. Such objects
-/// may be converted to `Message`s without any error paths.
+/// Trait describing something that promises to be a 32-byte uniformly random number.
+///
+/// In particular, anything implementing this trait must have neglibile probability
+/// of being zero, overflowing the group order, or equalling any specific value.
+///
+/// Since version 0.29 this has been deprecated; users should instead implement
+/// `Into<Message>` for types that satisfy these properties.
```
Comment (Kixunil) [inline src/lib.rs:203]: I wonder when are we going to remove this entirely.

**Review by apoelstra — COMMENTED** [review]
Successfully ran local tests on cff0e2ece791b0e207d115968b2f6c3ced51f9da.

**Comment by apoelstra** [comment]
Yes. How long is the delay between stuff getting merged into cargo and it showing up in a rustc nightly? I have no idea.

**Review by apoelstra — COMMENTED** [review]
Successfully ran local tests on 9e08c871cf86534f61255502956d39a69d599003.

**Comment by Kixunil** [comment]
> How long is the delay between stuff getting merged into cargo and it showing up in a rustc nightly?

IIUC it's should be the following night - hence the name "nightly"

**Comment by apoelstra** [comment]
I think that only applies to `rustc` itself. Everything else has a delay. In particular clippy has a delay, and ISTR that one of their developers told us it was because the clippy changes would ship alongside the cargo changes.

**Comment by apoelstra** [comment]
Reminder to me to merge https://github.com/rust-bitcoin/rust-secp256k1/pull/737 after this gets in

**Comment by Kixunil** [comment]
Do you want to address my comments?

File: secp256k1-sys/src/lib.rs (hunk 1 above)
Comment (apoelstra) [reply secp256k1-sys/src/lib.rs:139]: Sure.

File: src/ellswift.rs (hunk 2 above)
Comment (apoelstra) [reply src/ellswift.rs:286]: https://github.com/rust-bitcoin/rust-secp256k1/issues/741

File: src/lib.rs (hunk 3 above)
Comment (apoelstra) [reply src/lib.rs:203]: It was deprecated in April. Let's give it at least a year (unless it winds up blocking 1.0). I'd give it 2. It's self-contained and not hurting anything.

**Comment by apoelstra** [comment]
Sure. Removed "likely to be" from the UB comment. Should be good to go now.

**Review by Kixunil — APPROVED** [review]
ACK 38106866c1c23ea2c76879cae2332ea80064711a

**Review by apoelstra — COMMENTED** [review]
Successfully ran local tests on 38106866c1c23ea2c76879cae2332ea80064711a.

**Pull request merged**